File: Data Structures & Algorithms/remove-element/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} val
     * @return {number}
     */
    removeElement(nums, val) {
        let new_size = 0;
        for (let i = 0; i< nums.length; i++) {
            if (nums[i] != val) {
                nums[new_size] = nums[i];
                new_size++;
            }
        }
        return new_size;
    }
}

File: Data Structures & Algorithms/first-missing-positive/submission-2.cpp
class Solution {
public:
    int firstMissingPositive(vector<int>& nums) {
        int size = nums.size();
        bool contains1 = false;
        for (int i = 0; i<size; i++) {
            if (nums[i] == 1) {
                contains1 = true;
            }

            if (nums[i] <= 0 || nums[i] > size) {
                nums[i] = 1;
            }
        }

        for (int i= 0; i< size; i++) {
            int number = abs(nums[i]);
            int index = number - 1;

            if (nums[index] < 0) {
                continue;
            }

            nums[index] *= -1;
        }

        if (contains1 == false) {
            return 1;
        }

        for (int i = 0; i<size; i++) {
            if (nums[i] > 0) {
                return i + 1;
            } 
        }

        return size + 1;
    }
};